lib/jekyll/commands/serve.rb

```diff
@@ -220,16 +224,17 @@ def webrick_opts(opts)
 
         private
         def start_up_webrick(opts, destination)
-          if opts["livereload"]
-            @reload_reactor.start(opts)
-          end
-
           @server = WEBrick::HTTPServer.new(webrick_opts(opts)).tap { |o| o.unmount("") }
           @server.mount(opts["baseurl"].to_s, Servlet, destination, file_handler_opts)
 
           Jekyll.logger.info "Server address:", server_address(@server, opts)
-          launch_browser @server, opts if opts["open_url"]
-          boot_or_detach @server, opts
+        end
+
+        private
+        def start_up_livereload(opts)
```

Comment:
I don't think this need to be a new method.. it's going to be called just once anyways..
So it better to simply *start livereload* directly in the `:process` method.

```ruby
def process(opts)
  [...]
  @reload_reactor.start(opts) if opts["livereload"]
  start_up_webrick(opts, destination)

  launch_browser(@server, opts) if opts["open_url"]
  boot_or_detach(@server, opts)
end
```

Reply:
I do not have any particular opinion, so I follow your suggestions.

https://github.com/jekyll/jekyll/pull/6889/commits/7596a20348fc2b3b3ff964a12b0d549e64f6ec20